tests/datasource/fluent/data_asset/test_sql_asset.py

```diff
@@ -0,0 +1,103 @@
+from unittest.mock import create_autospec
+
+import pytest
+
+from great_expectations.core.batch_definition import BatchDefinition
+from great_expectations.core.partitioners import (
+    PartitionerYear,
+    PartitionerYearAndMonth,
+    PartitionerYearAndMonthAndDay,
+)
+from great_expectations.datasource.fluent import SQLDatasource
+from great_expectations.datasource.fluent.sql_datasource import _SQLAsset
+
+
+@pytest.mark.unit
+def test_add_batch_definition_fluent_sql__add_batch_definition_whole_table():
+    # arrange
+    mock_datasource = create_autospec(SQLDatasource)
+    asset = _SQLAsset(
+        name="test_asset",
+    )
+    asset._datasource = mock_datasource  # same pattern Datasource uses to init Asset
```

Comment:
Wondering if it's worth it to make a fixture that returns the already set up (datasource, asset) tuple. It might not be. Just when I see comments like this where we need to explain something, I think it makes the benefit of the abstraction a bit higher.

Reply:
sure, added in [aced278](https://github.com/great-expectations/great_expectations/pull/9732/commits/aced2789ad0a4452365c764257facaa450c18412). I typically prefer tests to be all inline if possible for legibility, but i think it's pretty equivalent in this case.